book/src/controlling_system_execution/state-specific_dispatcher_external.md

```diff
@@ -0,0 +1,279 @@
+# How to Define State Dispatcher: External
+
+This guide explains how to define a state-specific `Dispatcher` whose `System`s are passed in externally, which means at the moment of state struct creation. This is used when the list of `System`s is determined by user choices at runtime. For simplicity sake we'll be using the `SimpleState` trait for our custom state. 
```

Comment:
I'm having trouble understanding the difference between `External` and `Internal`. Both create the final dispatcher in `on_start`, so I would consider them quite equal.

Imo this is missing a transition to showcase how the dispatchers can be modified from outside.

Reply:
I tried to modify the `state_dispatcher` example and it seems to be possible to pass a dispatcher for your state on creation, in a different state. See here: https://github.com/marot/amethyst/pull/1/files

Reply:
The external/internal examples already existed as separate guides... but I feel like they can probably be merged into one. I think in the end it comes down to the `State` having its own `Dispatcher` that is executing during the `State`s `update(..)` method. How the `Dispatcher` is configured and where it comes from is more a personal taste. I'll fix it up, thank you!